### PR TITLE
Chugga chugga choo choo

### PR DESCRIPTION
**Pull request by tyleryasaka**
This is ready for review / hopefully mergeable. 😃 

Implements the current client in [choo](https://github.com/choojs/choo). This should make the code more maintainable and understandable going forward.

Disclaimer: it's still a bit messy, might contain a bug or 2 (though I tried hard to make sure everything is working properly). Some things could be done in a more choo/functional way than they are being done right now. Lots of room for improvement, but this should be a decent starting point.

### CODE REVIEW

**Comment by tyleryasaka** [comment]
I've been enjoying learning how to use it. Really like it so far!

File: client-choo/components/public-key-card.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    const icon = avatar(profile.peerId)
+    let followBtn
+    if (profile.peerId === IpfsID.identity.profile.peerId) {
+      followBtn = { disabled: 'disabled', label: '(You)' }
+    } else if (profile.following) {
+      followBtn = { disabled: '', label: OP_UNFOLLOW }
+    } else {
+      followBtn = { disabled: '', label: OP_FOLLOW }
+    }
+    // had to do this because of a really weird bug in the bundler, where it was lower-casing the `followBtn` var to `followbtn`
```
Comment (daviddahl) [inline client-choo/components/public-key-card.js]: was that a Browserify thing? was there an issue in their project repo?

File: client-choo/img/key.svg [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (daviddahl) [inline client-choo/img/key.svg]: I *think* this key svg is unneeded

File: client-choo/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+var css = require('sheetify')
+var choo = require('choo')
+
+css('tachyons')
```
Comment (daviddahl) [inline client-choo/index.js]: cssify. cool!

File: client-choo/components/proof.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+          tab.classList.add(klass)
+        })
+      }
+      console.log('highlight', tab.classList)
+    })
+  }
+
+  viewProof (event) {
+    var hash = event.target.dataset.hash
+    // new ProofDetail('proof-detail', { proofId: hash })
```
Comment (daviddahl) [inline client-choo/components/proof.js]: `viewProof` seems to be the only thing not working

File: client-choo/components/public-key-card.js (hunk 1 above)
Comment (tyleryasaka) [reply client-choo/components/public-key-card.js]: I didn't dig into it. I just looked at the bundle.js file and it had a lower-cased version of the variable. I searched the codebase and it was always camel-cased in the source. Not sure why that was happening.

**Comment by daviddahl** [comment]
Ok, I have done a *test* merge of master into this branch and tested all of the UI, including the proof detail / save URL bits. All is working now

**Comment by tyleryasaka** [comment]
awesome! 🚂🚋🚋🚋

**Comment by daviddahl** [comment]
Awesome work @tyleryasaka !!

**Pull request merged**